core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -969,6 +970,71 @@ class PlaintextConsumerTest extends BaseConsumerTest {
     }
   }
 
+  @Test
+  def testCoopearativeAssigorAndRejoin(): Unit = {
+    // 2 consumer using cooperative-sticky assignment
+    this.consumerConfig.setProperty(ConsumerConfig.GROUP_ID_CONFIG, "cooperative-sticky-group")
+    this.consumerConfig.setProperty(ConsumerConfig.PARTITION_ASSIGNMENT_STRATEGY_CONFIG, classOf[CooperativeStickyAssignor].getName)
+    val consumer1 = createConsumer()
+    val consumer2 = createConsumer()
+
+    // create a new topic, have 2 partitions
+    val topic1 = "topic1"
+    val producer = createProducer()
+    val expectedAssignment = createTopicAndSendRecords(producer, topic1, 2, 100)
+
+    assertEquals(0, consumer1.assignment().size)
+    assertEquals(0, consumer2.assignment().size)
+
+    val lock1 = new ReentrantLock()
+    var generationId1 = 0
+    var memberId1 = ""
+    val customRebalanceListener = new ConsumerRebalanceListener {
+      override def onPartitionsRevoked(partitions: util.Collection[TopicPartition]): Unit = {
+      }
+      override def onPartitionsAssigned(partitions: util.Collection[TopicPartition]): Unit = {
+        lock1.lock()
```

Comment:
I know in this simple test, there should not be deadlock happen, but I still think we should avoid any possibility here. Could we use `tryLock` instead of `lock`, to avoid to block the test forever?

Reply:
Same comments to below lock